api/scenario_api_utils/CMakeLists.txt

```diff
@@ -1,62 +1,28 @@
-cmake_minimum_required(VERSION 2.8.3)
+cmake_minimum_required(VERSION 3.5)
 project(scenario_api_utils)
 
-add_compile_options(-std=c++14)
+if(NOT CMAKE_CXX_STANDARD)
+  set(CMAKE_CXX_STANDARD 14)
+endif()
 
-find_package(catkin REQUIRED COMPONENTS
-  roscpp
-  geometry_msgs
-  tf2_ros
-  tf2_geometry_msgs
-)
+find_package(ament_cmake_auto REQUIRED)
+ament_auto_find_build_dependencies()
 
 find_package(yaml-cpp REQUIRED)
 
-catkin_package(
-  INCLUDE_DIRS include
-  LIBRARIES scenario_api_utils
-  CATKIN_DEPENDS roscpp geometry_msgs tf2_ros tf2_geometry_msgs
-#  DEPENDS system_lib
-)
-
-include_directories(
-  include
-  ${catkin_INCLUDE_DIRS}
-)
-
-add_library(scenario_api_utils SHARED
+ament_auto_add_library(scenario_api_utils SHARED
   src/scenario_api_utils.cpp
 )
-
-add_dependencies(scenario_api_utils
-  ${${PROJECT_NAME}_EXPORTED_TARGETS}
-  ${catkin_EXPORTED_TARGETS})
-
 target_link_libraries(scenario_api_utils
-  ${catkin_LIBRARIES}
   ${YAML_CPP_LIBRARIES}
 )
 
-install(DIRECTORY include/
-  DESTINATION ${CATKIN_PACKAGE_INCLUDE_DESTINATION}
-)
-
-install(TARGETS scenario_api_utils
-  ARCHIVE DESTINATION ${CATKIN_PACKAGE_LIB_DESTINATION}
-  LIBRARY DESTINATION ${CATKIN_PACKAGE_LIB_DESTINATION}
-  RUNTIME DESTINATION ${CATKIN_GLOBAL_BIN_DESTINATION}
-)
-
-install(DIRECTORY include/${PROJECT_NAME}/
-  DESTINATION ${CATKIN_PACKAGE_INCLUDE_DESTINATION}
-)
-
-catkin_install_python(
-  PROGRAMS scripts/ros_time_publisher.py
-  DESTINATION ${CATKIN_PACKAGE_BIN_DESTINATION}
+# Install python script  
+install(PROGRAMS scripts/ros_time_publisher.py
+  DESTINATION lib/${PROJECT_NAME}
 )
 
-install(
-  DIRECTORY launch
-  DESTINATION ${CATKIN_PACKAGE_SHARE_DESTINATION}
-  )
+ament_auto_package(INSTALL_TO_SHARE
+  launch
+  scripts
```

Comment:
I don't think you need that. Otherwise ` ros_time_publisher.py` appears twice in

```
$ tree install/scenario_api_utils
install/scenario_api_utils
├── include
│   └── scenario_api_utils
│       └── scenario_api_utils.h
├── lib
│   ├── libscenario_api_utils.so
│   └── scenario_api_utils
│       └── ros_time_publisher.py
└── share
    ├── ament_index
    │   └── resource_index
    │       ├── package_run_dependencies
    │       │   └── scenario_api_utils
    │       ├── packages
    │       │   └── scenario_api_utils
    │       └── parent_prefix_path
    │           └── scenario_api_utils
    ├── colcon-core
    │   └── packages
    │       └── scenario_api_utils
    └── scenario_api_utils
        ├── cmake
        │   ├── ament_cmake_export_dependencies-extras.cmake
        │   ├── ament_cmake_export_include_directories-extras.cmake
        │   ├── ament_cmake_export_libraries-extras.cmake
        │   ├── scenario_api_utilsConfig.cmake
        │   └── scenario_api_utilsConfig-version.cmake
        ├── environment
        │   ├── ament_prefix_path.dsv
        │   ├── ament_prefix_path.sh
        │   ├── library_path.dsv
        │   ├── library_path.sh
        │   ├── path.dsv
        │   └── path.sh
        ├── hook
        │   ├── cmake_prefix_path.dsv
        │   ├── cmake_prefix_path.ps1
        │   ├── cmake_prefix_path.sh
        │   ├── ld_library_path_lib.dsv
        │   ├── ld_library_path_lib.ps1
        │   └── ld_library_path_lib.sh
        ├── launch
        │   └── clock_publisher.launch.xml
        ├── local_setup.bash
        ├── local_setup.dsv
        ├── local_setup.sh
        ├── local_setup.zsh
        ├── package.bash
        ├── package.dsv
        ├── package.ps1
        ├── package.sh
        ├── package.xml
        ├── package.zsh
        └── scripts
            └── ros_time_publisher.py

```

Reply:
It depends whether  `ros_time_publisher.py` will be run via `ros2 run` or not. If we want that, yes, install it in `lib/${PROJECT_NAME}`, if not, better in `scripts`. Nevertheless, not both.

Reply:
I removed `scripts` from `ament_auto_package` command.